NOTES.md

```diff
@@ -0,0 +1,235 @@
+  "points": 2,
+  "firstName": "Mireya",
+  "lastName": "Armstrong",
+  "username": "Fred32",
+  "email": "Ayden9@hotmail.com",
+  "address": {
+    "country": "Germany",
+    "city": "Berlin"
+  },
+  "createdAt": "2020-05-18T00:00:00.000Z",
+  "updatedAt": "2021-05-18T05:17:48.493Z"
+}
+```
+
+**del** => if deleted successfully you will recieve a response of the user as an object, everything else will throw an error.
+
+### /login
+
+**post** => _to be updated_ => expects the following request =>
+
+```
+{
+	"email": "the users email"
+}
+```
+
+The response will be the user as an object.
+
+## Books
+
+All book calls start with /books
+
+### /
+
+**get** => to be cleared after testing, responseds with an array of all available book objects
+**post** => expects the following minimal request =>
+
+```
+{
+	"title": "xmen",
+	"subtitle": "The Big X",
+	"authors": ["J. K. Rowling"],
+	"publishedDate": 2002,
+	"pages": 10,
+	"shape": "good condition",
+	"category": ["fantasy"],
+	"selectedFiles": "file 03",
+	"owner": "a valid user mongoose id"
+}
+```
+
+_city will default to berlin_
+
+you will recieve the following response: an object with "newBook": the book as an object && "addBookToUser": the user who is the owner of that book =>
+
+```
+{
+  "newBook": {
+    "authors": [
+      "J. K. Rowling"
+    ],
+    "isbn": [],
+    "city": "Berlin",
+    "category": [
+      "fantasy"
+    ],
+    "selectedFiles": [
+      "file 03"
+    ],
+    "reserved": false,
+    "interestedUsers": [],
+    "_id": "60a351257062560d179d7d97",
+    "title": "xmen",
+    "subtitle": "The Big X",
+    "publishedDate": "2002",
+    "pages": 10,
+    "shape": "good condition",
+    "owner": "60a25ada6d711a9e70faa165",
+    "createdAt": "2021-05-18T05:31:17.831Z",
+    "updatedAt": "2021-05-18T05:31:17.831Z"
+  },
+  "addBookToUser": {
+    "avatar": "https://cdn.fakercloud.com/avatars/jjshaw14_128.jpg",
+    "points": 0,
+    "booksToOffer": [
+      "60a25ada6d711a9e70faa16a",
+      "60a25adb6d711a9e70faa16c",
+      "60a25ae16d711a9e70faa177",
+      "60a350be7062560d179d7d96",
+      "60a351257062560d179d7d97"
+    ],
+    "booksToRemember": [
+      "60a25adc6d711a9e70faa16d"
+    ],
+    "booksInterestedIn": [
+      "60a25ae06d711a9e70faa176",
+      "60a25ae16d711a9e70faa178"
+    ],
+    "matches": [],
+    "_id": "60a25ada6d711a9e70faa165",
+    "firstName": "Rickie",
+    "lastName": "Kilback",
+    "username": "Willa.Monahan",
+    "email": "Carlo.Walker@gmail.com",
+    "address": {
+      "country": "Germany",
+      "city": "Berlin"
+    },
+    "createdAt": "2021-05-17T12:00:26.167Z",
+    "updatedAt": "2021-05-18T05:31:18.031Z"
+  }
+}
+```
+
+### /:id
+
+**get** => will recieve a response as the book as an object
+**put** => please refer to user/:id put as it works mostly the same way. If updating an array with the request =>
+
+```
+{
+	"isbn": "this item"
+}
+```
+
+This will change the _whole array_ to be one item, the outcome will be =>
+
```

Comment:
is it recommendet to do it like that? if it overwrites it, do we loose the loose the second isbn ?

Reply:
this is the way it happens, however we can change this with middle ware

Reply:
ok